cms/djangoapps/contentstore/rest_api/v1/serializers/xblock.py

```diff
@@ -1,15 +1,64 @@
 """
 API Serializers for xblocks
 """
+from rest_framework import serializers
+from .common import StrictSerializer
 
+class XblockSerializer(StrictSerializer):
```

Comment:
This class seems off to me. I foresee that as the API is used by more and more people, this class will grow and grow and grow, until it's a beast adding no value. (Put another way, if fails the open/close design principle).

Should we be considering distinct classes for HTML, video, and LTI blocks?

Reply:
Here's my thoughts:
I posed the question previously on whether we need to make sure the serializers cover potential user input. The answer was: no, for now it is good enough to get the demo app running. That is the intention here.
I added additional fields I found just because it was very easy and will make us have to adjust less in the future.
Now I agree with you completely, there's actually a big problem because we have a very big unknown number of xblocks, since they are pluggable. I assume that the fields for the xblocks are actually in the "fields" dictionary, which we don't validate, but I'll need to investigate.
If they're in the fields, as long as we don't validate nested objects, we're good for now. Now long-term we may actually want to validate this too, especially since those fields are passed through to underlying xblocks, so I want to avoid havoc when a wrong field is passed.
If they're extra arguments that are not in the "fields", we'd need to add this to validations.
We can't do that whenever they pop up. So do we write our own serializer classes for all xblocks that pop up?
That would be a lot of work. Instead, the way forward I imagine:
xblocks already have their own fields with types baked in, it seems. For example, `xmodule/video_block/video_xfields.py`. I also found that for other serializers, there's a mechanism to convert these fields into serializer fields: see `cms/djangoapps/contentstore/rest_api/v0/serializers/advanced_settings.py`.
Now as far as I can tell, other xblocks have these fields as well, but they may not be defined in a separate class. If necessary, we could pretty easily extract them to their own class just as in video_xfields.
The next step would be to update the formal xblock specification to require xblock authors to have this as separate class, if necessary.
The outcome would be that we could automatically construct a serializer from any xblock.
We should also investigate whether anyone has already solved this problem, to avoid duplicate work.
Why is this desirable? Because we really don't have much validation in the backend. Validation is something I think any backend endpoint should have (maybe except ajax endpoints), even if used only internally to be called by MFEs. This will prevent future problems.
Xblocks are the most prevalent and complex part of our application, so being able to serialize any xblock would be very helpful.

Reply:
Per our offline conversation on this topic, you will add a comment to the effect that the pattern of just adding more field names here is not to be followed. This is good for starters, but as more endpoints call for more fields, a redesign should occur.